### DISCUSSION TITLE
eslint import/no-extraneous-dependencies  not work in pnpm

### DISCUSSION THREAD

**Discussion by pincman** [pnpm/pnpm#3291] [Q&A]
yarn working fine
<img width="642" alt="Jietu20210329-053648" src="https://user-images.githubusercontent.com/74093067/112768987-505db000-9051-11eb-89d2-e2987e99b82d.png">
but fater pnpm install,not show any error
<img width="684" alt="Jietu20210329-053909" src="https://user-images.githubusercontent.com/74093067/112769004-666b7080-9051-11eb-8e4d-1373b6ed3193.png">

**Comment by zkochan** [comment]
Try `pnpm install --shamefully-hoist`

**Comment by kopach** [comment]
From rule documentation https://github.com/import-js/eslint-plugin-import/blob/91f944d8e4b1591aa45f7d1b734dc2a2fe9acc57/docs/rules/no-extraneous-dependencies.md?plain=1#L8
> Modules have to be installed for this rule to work.

Basically, this rule check whether some dependency is under `node_modules`. PNPM adds to the root of `node_modules` only dependencies which are explicitly added to `package.json` (unless you use `--shamefully-hoist` flag which would hoist everything like `yarn` or `npm`, but that would negate quite big benefit of PNPM)

That said, this rule might not be needed at all if `PNPM` used since most of the time your import just won't work if it's not installed under `node_modules` thus added to package.json

---
This rule can still be useful with `PNPM` to check that you depend on `dependencies` and not `devDependencies` in your source code
